client/my-sites/themes/helpers.ts

```diff
@@ -1,18 +1,34 @@
 import { mapValues } from 'lodash';
+import { BaseSyntheticEvent } from 'react';
 import titlecase from 'to-title-case';
 import { gaRecordEvent } from 'calypso/lib/analytics/ga';
 import { isMagnificentLocale } from 'calypso/lib/i18n-utils';
 
-export function trackClick( componentName, eventName, verb = 'click' ) {
+interface ThemeOption {
+	action?: () => void;
+	extendedLabel?: string;
+	getUrl?: () => string;
+	header?: string;
+	hideForTheme?: () => void;
+	icon?: string;
+	label?: string;
+	separator?: boolean;
+}
+
+interface ThemeOptions {
+	[ key: string ]: ThemeOption;
+}
```

Comment:
You can also use a helper type called `Record<K, V>` that does the same thing:

```suggestion
type ThemeOptions = Record<string, ThemeOption>
```

Do we really not know what `options` could be? Do we need to use that wildcard `string` type declaration?

Reply:
Ah! We _do_ know what `options` could be. I was trying to be smarter about not writing out every single key, but it sounds like we want to prefer being more explicit when possible.